system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -986,41 +986,26 @@ class CLdapConnection : implements ILdapConnection, public CInterface
         StringBuffer hostbuf;
         for (int numHosts=0; numHosts < m_ldapconfig->getHostCount(); numHosts++)
         {
-            m_ldapconfig->getLdapHost(hostbuf);
-
             for(int retries = 0; retries <= LDAPSEC_MAX_RETRIES; retries++)
             {
+                m_ldapconfig->getLdapHost(hostbuf);
                 rc = connect(hostbuf.str(), proto);
-                if(!LdapServerDown(rc) || retries > LDAPSEC_MAX_RETRIES)
-                    break;
-                sleep(LDAPSEC_RETRY_WAIT);
-                if(retries < LDAPSEC_MAX_RETRIES)
-                    DBGLOG("Server temporarily unreachable, retrying ...");
-            }
-
-            if(rc == LDAP_SERVER_DOWN)
-            {
-                StringBuffer dc;
-                LdapUtils::getDcName(m_ldapconfig->getDomain(), dc);
-                if(dc.length() > 0)
+                if(rc == LDAP_SUCCESS)
```

Comment:
Changed in a completely different way to the code above?  Why??

Reply:
The above code is executed at startup, and tries to connect to the first LDAP AD even while in the retry loop. This is the same as the legacy implementation. It will blacklist it after retrying. This code (connection pool) needs to be sensitive to an AD going down at any point, even in the retry loop.  This is where Lisa will see the improvement

Reply:
Yes, but it is clearer if it is coded as:

```
               if (rc == LDAP_SUCCESS)
                    return true;
                if(rc != LDAP_TIMEOUT || retries >= LDAPSEC_MAX_RETRIES)
                    break;
                sleep(LDAPSEC_RETRY_WAIT);
                DBGLOG("LDAP AD Server %s temporarily unreachable for user %s on port %d, retrying...", hostbuf.str(), sysUserDN.str(), port);


```  
so the structure matches the structure of the 1st loop.